src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -3421,7 +3422,7 @@
 
     /**
      * '&amp;' symbol when used in a generic upper or lower bounds constrain
-     * e.g. {@code Comparable&lt;<? extends Serializable, CharSequence>}.
+     * e.g.&nbsp;{@code Comparable&lt;? extends Serializable, CharSequence&gt;}.
```

Comment:
Why the `&nbsp;`? It is between 2 printable characters so it shouldn't be needed. You usually only need it for multiple spaces in a row.
Shouldn't the brackets be ok since they inside the `{@code` tag?

Reply:
Unfortunately, nothing better can be done. Must be a way to understand where the _first_ sentence ends. For now it is "any period symbol, followed by a space or EOF".
Replacing the nbsp to a space will result the sentence ending in "e.g." which is not correct.

Reply:
>Shouldn't the brackets be ok since they inside the {@code tag?

They are ok. But please pay attention to the whole file uses &amp;lt;/&amp;gt; notation
```
The {@code &lt;&lt;=} (left shift assignment) operator.
```
No reason to have an exception here. Note also, that there is a typo:
```
Comparable&amp;lt;<
```
both "&amp;lt;" and "<" are present. This line should be fixed anyway.

Reply:
> both "&lt;" and "<" are present.

I see the issue now, thanks. This can be ignored.

Reply:
Unfortunately, it turns out that all encoded symbols inside {@code} remain encoded.
I started a new issue: #5423